### PR TITLE
Update existing telemetry Task

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR updates the existing `Collect Metrics and Logs` Task. It converts to the new-style CRD-based configuration.

### CODE REVIEW

File: _docs/tasks/metrics-logs.md [hunk 1]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
 as the example application throughout this task.
 
 ## Before you begin
-* [Install Istio](./installing-istio.html) in your kubernetes
-  cluster and deploy an application.
-
-* Configure your environment to support calling `istioctl mixer`.
-  This may require setting up port-forwarding for the Mixer Config API as described in the
-  [reference docs]({{home}}/docs/reference/commands/istioctl.html#istioctl-mixer) for `istioctl mixer`.
+* [Install Istio](./installing-istio.html) in your kubernetes cluster and deploy
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: nix kubernetes cluster. Lets keep the tasks generic.

File: _docs/tasks/metrics-logs.md [hunk 2]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
+  an application. This task assumes that Mixer is setup in a default
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: the Mixer

File: _docs/tasks/metrics-logs.md [hunk 3]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
+  configuration (`--configDefaultNamespace=istio-config-default`). If a
+  different value is used, the configuration in this Task (and commands to be
+  issued) must be updated to match.
 
-* Configure your environment to support accessing the Istio dashboard, as described in the 
-  [Installation Guide](./installing-istio.html). This requires installing the optional add-ons
-  ([Prometheus](https://prometheus.io) and [Grafana](https://grafana.com/)), as well as verifying access to
-  the dashboard. The Istio dashboard will be used to verify task success.
+* Install the optional add-on [Prometheus](https://prometheus.io). Prometheus
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: Can we use this opportunity to move that content from install task?

File: _docs/tasks/metrics-logs.md [hunk 4]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
 ## Before you begin
-* [Install Istio](./installing-istio.html) in your kubernetes
-  cluster and deploy an application.
-
-* Configure your environment to support calling `istioctl mixer`.
-  This may require setting up port-forwarding for the Mixer Config API as described in the
-  [reference docs]({{home}}/docs/reference/commands/istioctl.html#istioctl-mixer) for `istioctl mixer`.
+* [Install Istio](./installing-istio.html) in your kubernetes cluster and deploy
+  an application. This task assumes that Mixer is setup in a default
+  configuration (`--configDefaultNamespace=istio-config-default`). If a
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: do we have a namespace for istio config? isn't it istio-system ?

File: _docs/tasks/metrics-logs.md [hunk 5]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
+        - newlog.logentry
+   ---
    ```
 
-   The expected output is:
+1. Push the new configuration.
 
    ```bash
-   Error: the server could not find the requested resource
+   kubectl -n istio-config-default apply -f new_telemetry.yaml
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: istioctl

File: _docs/tasks/metrics-logs.md [hunk 6]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
    ```
 
-   For purposes of this task, please refresh the page several times or issue the curl
-   command a few times to generate traffic.
+   For purposes of this task, please refresh the page several times or issue the
+   curl command a few times to generate traffic.
+
+1. Verify that the new metric values are being generated and collected.
+
+   Setup port-forwarding for Prometheus:
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: In a Kubernetes environment, setup ... using the following command:

File: _docs/tasks/metrics-logs.md [hunk 7]
```diff
@@ -9,210 +10,300 @@ type: markdown
...
-   kubectl port-forward $(kubectl get pod -l app=grafana -o jsonpath='{.items[0].metadata.name}') 3000:3000 &
+   kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=prometheus -o jsonpath='{.items[0].metadata.name}') 9090:9090 &
    ```
 
-   Then open the Istio dashboard in a web browser: [http://localhost:3000/dashboard/db/istio-dashboard](http://localhost:3000/dashboard/db/istio-dashboard)
-   
-   One of the rows in the dashboard will be named "reviews". If that row is not visible, please refresh the dashboard page. The "reviews" row
-   contains a graph entitled "Response Size by Source And Version". The graph displays a breakdown of the distribution of Response Sizes returned
-   by the "reviews" service.
+   Then open [this
```
Comment (rshriram) [inline _docs/tasks/metrics-logs.md]: You can now view the number of requests via this link.. 
add explanation as another sentence.

**Comment by rshriram** [comment]
Also the new path is samples/bookinfo/kube/..

File: _docs/tasks/metrics-logs.md (hunk 4 above)
Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: Mixer, by default, uses `istio-config-default`. It is configurable, but using `istio-system` is potentially problematic, iirc (issue: how to specify configuration just for the istio components themselves, but not affect global configuration).

**Comment by rshriram** [comment]
Per offline convo, grafana/service graph will be a separate task.

File: _docs/tasks/metrics-logs.md (hunk 5 above)
Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: istioctl doesn't support mixer rules, as far as i know.

Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: well, no, it turns out my version was just old enough not to support it.

File: _docs/tasks/metrics-logs.md (hunk 3 above)
Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: As discussed offline, i'm going to create a "Using Addons" or similar task.

File: _docs/tasks/metrics-logs.md (hunk 6 above)
Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: done.

File: _docs/tasks/metrics-logs.md (hunk 7 above)
Comment (douglas-reid) [reply _docs/tasks/metrics-logs.md]: Added some text that hopefully clarifies.

**Comment by rshriram** [comment]
@douglas-reid ^^^

**Comment by douglas-reid** [comment]
@Rachael-Graham thanks for the great review. I'll make changes in a follow-on PR and send them to you for approval.

**Pull request merged**